src/server/report/test.js

```diff
@@ -339,83 +268,20 @@ class Test extends LitElement {
 			</div>
 		`;
 	}
-	_renderPillbox(name, selectedValue, items) {
-		const renderItem = (i) => {
-			const id = getId();
-			return html`
-				<div class="pill">
-					<input type="radio" id="${id}" name="${name}" value="${i.value}" @change="${this._handlePillboxChange}" ?checked="${i.value === selectedValue}">
-					<label for="${id}">${i.icon}${i.label}</label>
-				</div>
-			`;
-		};
-		return html`<div class="pill-box">${items.map(i => renderItem(i))}</div>`;
+	_renderTabButtons(tabs) {
+		if (tabs.length < 2) return nothing;
+		return renderTabButtons('browser results', tabs, index => this._selectedBrowserIndex = index);
 	}
-	_renderTabButtons(browsers, selectedBrowser, testData) {
-
-		const onKeyDown = (e) => {
-			let focusOn;
-			switch (e.key) {
-				case 'ArrowRight':
-					focusOn = e.target.nextElementSibling || e.target.parentNode.firstElementChild;
-					break;
-				case 'ArrowLeft':
-					focusOn = e.target.previousElementSibling || e.target.parentNode.lastElementChild;
-					break;
-				case 'Home':
-					focusOn = e.target.parentNode.firstElementChild;
-					break;
-				case 'End':
-					focusOn = e.target.parentNode.lastElementChild;
-					break;
-			}
-			if (focusOn) focusOn.focus();
-		};
+	_renderTabPanels(tabs) {
 
-		const renderTabButton = (browser, index) => {
-			const result = testData.results.find(r => r.name === browser);
-			const selected = (browser === selectedBrowser.name);
-			const status = result.passed ? 'passed' : 'failed';
-			const onClick = () => {
-				return () => this._selectedBrowserIndex = index;
-			};
-			const statusClass = {
-				pass: result.passed,
-				error: !result.passed
-			};
-			return html`
-				<button
-					aria-controls="tabpanel-${browser}"
-					aria-selected="${selected ? 'true' : 'false'}"
-					id="tab-${browser}"
-					role="tab"
-					tabindex="${selected ? '0' : '-1'}"
-					type="button"
-					@click="${onClick()}">
-						<span>
-							${browser} <span class="${classMap(statusClass)}">(${status})</span>
-						</span>${selected ? html`
-						<div class="tab-selected-indicator"></div>` : nothing}
-				</button>`;
-		};
-
-		return html`
-			<div role="tablist" aria-label="browser results" @keydown="${onKeyDown}">
-				${browsers.map((b, i) => renderTabButton(b.name, i))}
-			</div>
-		`;
-	}
-	_renderTabPanels(browsers, selectedBrowser, testData) {
-
-		const renderTabPanel = (browser) => {
-			return html`
-				<div id="tabpanel-${browser}" role="tabpanel" aria-labelledby="tab-${browser}" ?hidden="${browser !== selectedBrowser.name}">
-					${this._renderTestResults(browser, testData)}
-				</div>
-			`;
-		};
+		let panelsContent;
+		if (tabs.length < 2) {
+			panelsContent = tabs.map(t => t.content);
+		} else {
+			panelsContent = tabs.map(t => renderTabPanel(t));
+		}
```

Comment:
Slight bug fix here -- if there's only 1 browser showing we don't render the tabs at all, so this also skips rendering the surrounding panel.